djangocms_versioning/admin.py

```diff
@@ -135,6 +141,176 @@ def has_change_permission(self, request, obj=None):
+        Return the state of a given version
+        """
+        return self.get_version(obj).get_state_display()
+
+    get_versioning_state.short_description = _("State")
+
+    def get_author(self, obj):
+        """
+        Return the author who created a version
+        :param obj: Versioned content model Instance
+        :return: Author
+        """
+        return self.get_version(obj).created_by
+
+    get_author.short_description = _("Author")
+
+    def get_modified_date(self, obj):
+        """
+        Get the last modified date of a version
+        :param obj: Versioned content model Instance
+        :return: Modified Date
+        """
+        return self.get_version(obj).modified
+
+    get_modified_date.short_description = _("Modified")
+
+    def get_list_display_configuration(self):
+        """
+        Return the configured list display from cms config versionable object, if it is not configured,
+        return the default django value if list_display is not set.
+        """
+        versionable = versionables.for_content(self.model)
+        return versionable.admin_list_display_fields or ["__str__"]
+
+    def get_preview_method_configuration(self):
+        """
+        Return the preview method if available, otherwise return None
+        :return: method or None
+        """
+        versionable = versionables.for_content(self.model)
+        return versionable.preview_url or None
+
+    def _list_actions(self, request):
+        """
+        A closure that makes it possible to pass request object to
+        list action button functions.
+        """
+
+        def list_actions(obj):
+            """Display links to state change endpoints
+            """
+            return format_html_join(
+                "",
+                "{}",
+                ((action(obj, request),) for action in self.get_list_actions()),
+            )
+
+        list_actions.short_description = _("actions")
+        return list_actions
+
+    def get_list_display(self, request):
+        # Get list display from app_config
+        list_display = self.get_list_display_configuration()
+        versioning_list_display_items = ["get_author", "get_modified_date", "get_versioning_state"]
+
+        # Ensure fields aren't already within list_display
+        if not set(versioning_list_display_items).issubset(list_display):
+            list_display.extend(versioning_list_display_items)
+            # Add actions last
+            list_display.extend([self._list_actions(request)])
+        return list_display
+
+    def _get_preview_link(self, obj, request, disabled=False):
+        """
+        Return a user friendly button for previewing the content model
+        :param obj: Instance of versioned content model
+        :param request: The request to admin menu
+        :param disabled: Should the link be marked disabled?
+        :return: Preview icon template
+        """
+        preview_url = self.get_preview_method_configuration()
```

Comment:
```suggestion
        preview_url = self._get_preview_url(obj)
```
If accepting the changes above ^^^